### DISCUSSION TITLE
How to switch chart type from one to another one?

### DISCUSSION THREAD

**Discussion by timocov** [tradingview/lightweight-charts#634] [Q&A] [answered]
Is it possible to switch series type from one type (e.g. line) to another one (e.g. bar)?

**Comment by timocov** [comment, marked as answer]
As soon as series's type directly affects type of the data (single-value vs ohlc), type of the options and so on, it is not possible to seamlessly switch series from one type to another one. If you'd like to do it, you need to create the new series with desired type and remove the old one.

**Comment by iphyman** [comment]
@timocov it's 2024, is it possible to switch between charts, I have OHLC data works fine when on candle chart but switching to series and calling `series.setData` returns undefined value error ?
